dht-node/src/dht_node/index.ts

```diff
@@ -190,20 +198,69 @@ async function writeFederatedHomeCommunityEnries(pubKey: any): Promise<Community
+    for (let i = 0; i < homeApiVersions.length; i++) {
+      const homeCom = DbFederatedCommunity.create()
       homeCom.foreign = false
-      homeCom.apiVersion = homeApi.api
-      homeCom.endPoint = homeApi.url
-      homeCom.publicKey = pubKey.toString('hex')
-
-      // this will NOT update the updatedAt column, to distingue between a normal update and the last announcement
+      homeCom.apiVersion = homeApiVersions[i].api
+      homeCom.endPoint = homeApiVersions[i].url
+      homeCom.publicKey = Buffer.from(pubKey)
       await DbFederatedCommunity.insert(homeCom)
-      logger.info(`federation home-community inserted successfully: ${JSON.stringify(homeCom)}`)
-    })
+      logger.info(
+        `federation home-community inserted successfully: ${JSON.stringify(homeApiVersions[i])}`,
+      )
+    }
   } catch (err) {
-    throw new Error(`Federation: Error writing HomeCommunity-Entries: ${err}`)
+    throw new Error(`Federation: Error writing federated HomeCommunity-Entries: ${err}`)
   }
   return homeApiVersions
 }
+
+export async function writeHomeCommunityEntry(pubKey: string): Promise<void> {
+  try {
+    // check for existing homeCommunity entry
+    let homeCom = await DbCommunity.findOne({
+      foreign: false,
+      publicKey: Buffer.from(pubKey),
+    })
+    if (!homeCom) {
+      // check if a homecommunity with a different publicKey still exists
+      homeCom = await DbCommunity.findOne({ foreign: false })
+    }
+    if (homeCom) {
+      // simply update the existing entry, but it MUST keep the ID and UUID because of possible relations
+      homeCom.publicKey = Buffer.from(pubKey)
+      homeCom.url = CONFIG.FEDERATION_COMMUNITY_URL + '/api/'
+      homeCom.name = CONFIG.COMMUNITY_NAME
+      homeCom.description = CONFIG.COMMUNITY_DESCRIPTION
+      await DbCommunity.save(homeCom)
+      logger.info(`home-community updated successfully: ${JSON.stringify(homeCom)}`)
+    } else {
+      // insert a new homecommunity entry including a new ID and a new but ensured unique UUID
+      homeCom = new DbCommunity()
+      homeCom.foreign = false
+      homeCom.publicKey = Buffer.from(pubKey)
+      homeCom.communityUuid = await newCommunityUuid()
+      homeCom.url = CONFIG.FEDERATION_COMMUNITY_URL + '/api/'
+      homeCom.name = CONFIG.COMMUNITY_NAME
+      homeCom.description = CONFIG.COMMUNITY_DESCRIPTION
+      homeCom.creationDate = new Date()
+      await DbCommunity.insert(homeCom)
+      logger.info(`home-community inserted successfully: ${JSON.stringify(homeCom)}`)
+    }
+  } catch (err) {
+    throw new Error(`Federation: Error writing HomeCommunity-Entry: ${err}`)
+  }
+}
+
+const newCommunityUuid = async (): Promise<string> => {
+  let uuid: string
+  let countIds: number
+  do {
+    uuid = uuidv4()
+    countIds = await DbCommunity.count({ where: { communityUuid: uuid } })
+    if (countIds > 0) {
+      logger.info('CommunityUuid creation conflict...')
+    }
+  } while (countIds > 0)
+  return uuid
```

Comment:
you can return within the while and avoid all those variables pretty much.

```suggestion
  while(true){
    const communityUuid = uuidv4()
    if(await DbCommunity.count({ where: { communityUuid } }) === 0){
      return communityUuid
    }
  }
```

Further read: https://stackoverflow.com/questions/32170943/javascript-while-loop-return

Reply:
yes i know, but with my solution we get logging information about the amount of conflicts of uuid generation to see if it will happen and how often.